File: http-tunnel/http-dyn/python/python2/requests/proxy-demo.py
#! -*- encoding:utf-8 -*-

import requests

# 要访问的目标页面
targetUrl = "http://test.abuyun.com"

# 代理服务器
proxyHost = "http-dyn.abuyun.com"
proxyPort = "9020"

# 代理隧道验证信息
proxyUser = "H01234567890123D"
proxyPass = "0123456789012345"

proxyMeta = "http://%(user)s:%(pass)s@%(host)s:%(port)s" % {
    "host" : proxyHost,
    "port" : proxyPort,
    "user" : proxyUser,
    "pass" : proxyPass,
}

proxies = {
    "http"  : proxyMeta,
    "https" : proxyMeta,
}

resp = requests.get(targetUrl, proxies=proxies)

print resp.status_code
print resp.text

File: http-tunnel/http-dyn/nodejs/request/proxy-demo.js
"use strict";

const request = require("request");

// 要访问的目标页面
const targetUrl = "http://test.abuyun.com";

// 代理服务器
const proxyHost = "http-dyn.abuyun.com";
const proxyPort = 9020;

// 代理隧道验证信息
const proxyUser = "H01234567890123D";
const proxyPass = "0123456789012345";

const proxyUrl = "http://" + proxyUser + ":" + proxyPass + "@" + proxyHost + ":" + proxyPort;

const proxiedRequest = request.defaults({"proxy": proxyUrl});

const options = {
    url     : targetUrl,
    headers : {
    }
};

proxiedRequest
    .get(options, function (err, res, body) {
        console.log("got response: " + res.statusCode);
    })
    .on("error", function (err) {
        console.log(err);
    })
;
